src/rust/storage/seg/src/ttl_buckets/ttl_bucket.rs

```diff
@@ -106,6 +106,34 @@ impl TtlBucket {
         }
     }
 
+    // clear segments from this TtlBucket, returns the number of segments expired
+    pub(super) fn clear(&mut self, hashtable: &mut HashTable, segments: &mut Segments) -> usize {
```

Comment:
I personally would prefer `bulk_clear` over `clear` for all the new `clear` traits, and keep the original `segment.clear` as the only `clear` function.

Reply:
Happy to make the change if you prefer. I was trying to keep the high-level API closer to the std library HashMap: https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.clear

Reply:
Ah I see. this is fine